coverage/docs/design.md

```diff
@@ -0,0 +1,138 @@
+Produces per-file and per-package coverage result as input for TestGrid. Testgrid can use the data produced here to display coverage trend in a tabular or graphical way. 
+1. A PR is merged
+2. Post-submit job started
+3. Generate coverage profile. Completion marker generated upon successful run. Both stored
+ in artifacts directory.
+    - Completion marker is used by later pre-submit job when searching for a healthy and complete 
+    code coverage profile in the post-submit jobs
+    - Successfully generated coverage profile may be used as the basis of comparison for coverage change, 
+    as mentioned in pre-submit workflow
+4. Generate / store per-file coverage data
+    - Stores in the XML format, that is used by TestGrid, and dump it in artifacts directory
+    - The junit_bazel.xml should be a valid junit xml file. See 
+[JUnit XML format](https://www.ibm.com/support/knowledgecenter/en/SSQ2R2_14.1.0/com.ibm.rsar.analysis.codereview.cobol.doc/topics/cac_useresults_junit.html)
+    - For each file that has a coverage level lower than the threshold, the corresponding entry in the xml should have a \<failure\> tag
+
+## Locally running presubmit and post-submit workflows
+Both workflows may be triggered locally in command line, as long as all the required flags are 
+supplied correctly. In addition, the following env var needs to be set:
+- JOB_TYPE (one of 'presubmit', 'postsubmit', 'local-presubmit')
+
+use 'local-presubmit' will run the presubmit workflow without posting result on github PR
+  
+## Covbot
+As mentioned in the presubmit workflow section, covbot is the short name for the robot github 
+account used to report code coverage change results. It can be created as a regular github 
+account - it does not need to be named covbot as that name is already taken on Github. It only need a 
+comment access to the repo it need to be run on. If the repo is private, it also need read access. 
+  
+After the robot account is created, download the github token and supply the path to the token 
+file to code coverage binary, as the value for parameter "github-token"
+
+
+# Usage with prow
+Prow can be used as the system to handle Github events mentioned in the two workflows. 
+
+Prow, in both workflows, supplies the flags & secrets for the binary, clones the repository and uploads logs & artifacts to GCS bucket.
+
+The pre-submit prow job is triggered by any new commit to a PR. At the end of the binary run, it can return a pass or fail status context to Github. Tide can use that status to block PR with low coverage.
+
+The post-submit prow job is triggered by merge events to the base branch.
+
+## Prow Configuration File
+As mentioned earlier, we use configuration file to store repository specific information. Below is an example that contains the args that will be supplied to the coverage container
+```
+  - name: pull-knative-serving-go-coverage
+    labels:
+      preset-service-account: "true"
```

Comment:
This preset is meaningless without the its definition. This will be much clearer if you just add the volumes and env vars that the preset provides.